proposals/extensions.md

```diff
@@ -50,6 +50,48 @@ receiver_parameter // add
 Extension declarations shall only be declared in non-generic, non-nested static classes.  
 It is an error for a type to be named `extension`.  
 
+### Declaration spaces
+
+Section [7.3 Declarations](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/basic-concepts.md#73-declarations) in the C# Standard is updated as follows (additions in **bold non-italic**):
+
+> ...
+> 
+> There are several different types of declaration spaces, as described in the following.
+> 
+> ...
+> 
+> - Each method declaration, property declaration, property accessor declaration, indexer declaration, indexer accessor declaration, operator declaration, instance constructor declaration, anonymous function, and local function creates a new declaration space called a ***local variable declaration space***. Names are introduced into this declaration space through formal parameters (*fixed_parameter*s and *parameter_array*s) and *type_parameter*s. The set accessor for a property or an indexer introduces the name `value` as a formal parameter. 
+**An *extension_declaration* introduces to each member declaration directly contained within it its *type_parameter_list* as type parameters and the *identifier*, if any, of its *receiver_parameter* as a formal parameter.**
+
+There is an existing need to specify (maybe in Section[12.8.4 Simple Names](https://github.com/dotnet/csharpstandard/blob/standard-v7/standard/expressions.md#1284-simple-names)) something along the lines of the following:
+
+> If a local variable directly occurring in a given declaration space is accessed from within a static member or static lambda expression that is itself nested within that declaration space, a compile-time error is given.
+
+Thus, for receiver parameters as well as any other local variable, a reference from within a static context still resolves to that variable, but leads to an error.
+
+``` c#
+public static class E
+{
+    extension(string s)
+    {
+        public int M1(int i) // *1*
+        {
+            return s.Length + i;
+        }
+        public int M2(string s) // *2* Error: Cannot reuse name `s`
```

Comment:
> // *2* Error: Cannot reuse name `s`

Consider adding a similar example with type parameters. Something like:
```
public static class E
{
    extension<T>(T[] s)
    {
        public int M2<T>() // *2* Is this an error or a warning?
```

Reply:
Same for the following?

```cs
public static class E<T>
{
    extension<T>(T[] s) // *2* Is this an error or a warning?
    {
        // ...
    }
}
```

Reply:
> Same for the following?

Enclosing type cannot be generic

Reply:
Good idea. Added type parameter example.